monkey/monkey_island/cc/services/config.py

```diff
@@ -538,12 +539,26 @@ def _format_icmp_scan_from_flat_config(config: Dict):
     @staticmethod
     def _format_fingerprinters_from_flat_config(config: Dict):
         flat_fingerprinter_classes_field = "finger_classes"
+        flat_http_ports_field = "HTTP_PORTS"
 
-        formatted_fingerprinters = config[flat_fingerprinter_classes_field]
-        config.pop(flat_fingerprinter_classes_field)
+        formatted_fingerprinters = [
+            {"name": f, "options": {}} for f in sorted(config[flat_fingerprinter_classes_field])
+        ]
+
+        if "HTTPFinger" in config[flat_fingerprinter_classes_field]:
+            for fp in formatted_fingerprinters:
+                if fp["name"] == "HTTPFinger":
+                    fp["options"] = {"http_ports": sorted(config[flat_http_ports_field])}
```

Comment:
Move this to a separate function?

Reply:
I can if you feel strongly about it. This code will be deleted, hopefully in the not-to-distant future.